### PR TITLE
Pull request for issue #82

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/test/java/fi/helsinki/cs/tmc/core/commands/DownloadExercisesTest.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ public void downloadAllExercises() throws Exception {
         List<Exercise> exercises = download.get();
         Path exercisePath = folder.resolve("2013_ohpeJaOhja/viikko1/Viikko1_001.Nimi");
 
-        assertEquals(exercises.size(), 152);
```
Comment (UMTti) [inline src/test/java/fi/helsinki/cs/tmc/core/commands/DownloadExercisesTest.java]: Why this?

**Comment by ghost** [comment]
Because the test case uses course.json, which has 153 exercises instead of 152.
The test case previously omitted one exercise.

File: src/main/java/fi/helsinki/cs/tmc/core/communication/TmcApi.java [hunk 2]
```diff
@@ -167,13 +169,14 @@ public Course getCourseFromString(String jsonString) {
         Gson mapper = new Gson();
         Course course = mapper.fromJson(courseJson.getAsJsonObject("course"), Course.class);
 
+        if (course == null) {
+            return Optional.absent();
+        }
+
```
Comment (ljleppan) [inline src/main/java/fi/helsinki/cs/tmc/core/communication/TmcApi.java:176]: Can we replace this `if` with  [`Optional.fromNullable(T)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Optional.html#fromNullable%28T%29) ?

**Comment by jamo** [comment]
Nice work, please rebase with our master branch so that this could be merged.

**Comment by ljleppan** [comment]
LGTM

**Comment by ghost** [comment]
It's ready to be merged now.

**Comment by ljleppan** [comment]
As the master has moved on, this needs to be (re?)rebased before we can merge.

**Comment by ghost** [comment]
I synced the branch again with master.

**Pull request merged**